Examples/Algorithms/TrackFitting/include/ActsExamples/TrackFitting/TrackFittingAlgorithm.hpp

```diff
@@ -92,6 +93,8 @@ class TrackFittingAlgorithm final : public BareAlgorithm {
     bool multipleScattering = true;
     /// Some more detailed steering - correct for e-loss
     bool energyLoss = true;
+    // Correction for non-linearity effect during transform from free to bound
+    Acts::detail::FreeToBoundCorrection freeToBoundCorrection = false;
```

Comment:
Honestly, I don't like this auto-construction from a boolean. We've had this in various spots in the code, but my understanding was we want to try to reduce this. Personally, I think an explicit constructor (so you have to type out the name) with (potentially optional) arguments would be preferrable.

@asalzburger did you have an opinion on this pattern?

Reply:
Fine with me if that's the consensus, I found it useful for simple on/off flags, but I agree it can be misleading as it is not explicit.

So, we can remove the assignment operator and make a default constructor that sets the internal boolean to false.

Reply:
OK. Changed to use the constructor now.